packages/flutter/test/widgets/scrollbar_test.dart

```diff
@@ -248,6 +250,134 @@ void main() {
     },
   );
 
+  test(
+    'scrollbarOrientation are respected',
+    () {
+      const double viewportDimension = 23;
+      const double maxExtent = 100;
+      final ScrollMetrics startingMetrics = defaultMetrics.copyWith(
+        maxScrollExtent: maxExtent,
+        viewportDimension: viewportDimension,
+      );
+      const Size size = Size(600, viewportDimension);
+      const double margin = 0;
+
+      for (final ScrollbarOrientation scrollbarOrientation in ScrollbarOrientation.values) {
+        final AxisDirection axisDirection;
+        if (scrollbarOrientation == ScrollbarOrientation.left || scrollbarOrientation == ScrollbarOrientation.right)
+          axisDirection = AxisDirection.down;
+        else
+          axisDirection = AxisDirection.right;
+
+        painter = _buildPainter(
+          crossAxisMargin: margin,
+          scrollMetrics: startingMetrics,
+          scrollbarOrientation: scrollbarOrientation,
+        );
+
+        painter.update(
+          startingMetrics.copyWith(axisDirection: axisDirection),
+          axisDirection
+        );
+
+        painter.paint(testCanvas, size);
+        final Rect rect = captureRect();
+
+        switch (scrollbarOrientation) {
+          case ScrollbarOrientation.left:
+            expect(rect.left, 0);
+            expect(rect.top, 0);
+            expect(rect.right, _kThickness);
+            expect(rect.bottom, _kMinThumbExtent);
+            break;
+          case ScrollbarOrientation.right:
+            expect(rect.left, 600 - _kThickness);
+            expect(rect.top, 0);
+            expect(rect.right, 600);
+            expect(rect.bottom, _kMinThumbExtent);
+            break;
+          case ScrollbarOrientation.top:
+            expect(rect.left, 0);
+            expect(rect.top, 0);
+            expect(rect.right, _kMinThumbExtent);
+            expect(rect.bottom, _kThickness);
+            break;
+          case ScrollbarOrientation.bottom:
+            expect(rect.left, 0);
+            expect(rect.top, 23 - _kThickness);
+            expect(rect.right, _kMinThumbExtent);
+            expect(rect.bottom, 23);
+            break;
+        }
+      }
+    },
+  );
+
+  test(
+    'scrollbarOrientation default values are correct',
+    () {
```

Comment:
Formatting nit here and elsewhere

```suggestion
  test('scrollbarOrientation default values are correct', () {
```